lib/sp_configuration.rb

```diff
@@ -3,7 +3,6 @@
 module AmzSpApi
   class SpConfiguration < Configuration
     attr_accessor :refresh_token, :client_id, :client_secret, :sandbox, :region,
-      :aws_access_key_id, :aws_secret_access_key, :credentials_provider, # either access key or credentials_provider for AWS Signer, e.g. Aws::STS::Client
```

Comment:
these incompatible API changes would force this to be part of a major version change whereas if we left them in place and even better added this we could roll it out in a patch

```
  extend Gem::Deprecate
  deprecate :aws_access_key_id=, :none, 2025, 1
  deprecate :aws_secret_access_key=, :none, 2025, 1
  deprecate :credentials_provider=, :none, 2025, 1